packages/transactions/src/OutboundShipment/ListView/ListView.tsx

```diff
@@ -63,18 +42,54 @@ export const OutboundShipmentListView: FC = () => {
   });
   const { data: response, isLoading } = useQuery(
     ['transaction', 'list', queryProps],
-    () => queryFn(queryProps)
+    () => listQueryFn(queryProps),
+    { notifyOnChangeProps: ['data'] }
   );
 
   const queryClient = useQueryClient();
 
-  const { isLoading: mutationLoading, mutateAsync } = useMutation(deleteFn, {
+  const { mutateAsync: deleteMutateAsync } = useMutation(deleteFn, {
+    onSuccess: () => queryClient.invalidateQueries(['transaction']),
+  });
+
+  const { mutateAsync } = useMutation(updateTransactionFn, {
+    onMutate: patch => {
+      const key = ['transaction', 'list', queryProps];
+      const previousCached =
+        queryClient.getQueryData<{ data: Transaction[]; totalLength: number }>(
+          key
+        );
+
+      const previousData = [...(previousCached?.data ?? [])];
+
+      const existingRowIdx = previousData.findIndex(
+        ({ id }) => id === patch.id
+      );
+      previousData[existingRowIdx] = patch;
+
+      queryClient.setQueryData(key, { ...previousData, data: previousData });
+
+      return { previousCached };
+    },
```

Comment:
optimistically updating (and not handling the error case) - think there should be an easier way to do this. Likely need to do it for other things like deletion and any mutations on the `ListView` which aren't triggered by an `onSAve` event

Reply:
something like: Revert + snackbar to say "Nope!"

Reply:
yeah, feels like a big manual process. if nothing else, good to extract it out if we can move this into generic hooks. logic looks good though 🤷‍♂️